packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/task/s3_upload_task.dart

```diff
@@ -206,10 +206,12 @@ class S3UploadTask {
         _completeUploadWithError(error, stackTrace);
       }
 
-      if (_fileSize <= _minPartSize) {
-        // S3 multipart upload requires minimum part size to be 5MB. Multipart
-        // upload cannot be initiated for a file with size less than 5MB so use
-        // putObject
+      if (_fileSize > part_size_util.maxSingleObjectSize) {
+        _completeUploadWithError(S3Exception.uploadSourceIsTooLarge());
+        return;
+      }
+
+      if (_fileSize < part_size_util.minPartSize) {
```

Comment:
If _fileSize is unknown (`== -1`), wouldn't it make more sense to try a multipart upload and fallback to putobject if we reach the end before hitting the min part size?

Reply:
Change the `size` param back to required when instantiate with `AWSFile.fromStream`, which should eliminate this -1 implication.

We can add support of uploading from data stream with unknown size in the future if it's requested.